zenoh-jni/src/query.rs

```diff
@@ -74,12 +79,19 @@ pub(crate) unsafe extern "C" fn Java_io_zenoh_jni_JNIQuery_replySuccessViaJNI(
     qos: jbyte,
     attachment: JByteArray,
 ) {
-    let key_expr = Arc::from_raw(key_expr_ptr);
-    let key_expr_clone = key_expr.deref().clone();
-    std::mem::forget(key_expr);
+    let key_expr = match process_kotlin_key_expr(&mut env, &key_expr_str, key_expr_ptr) {
+        Ok(key_expr) => key_expr,
+        Err(err) => {
+            if let Err(err) = err.throw_on_jvm(&mut env) {
+                tracing::error!("{}", err);
+            }
+            return;
+        }
+    };
```

Comment:
i still think that there may be a better rust pattern for this. 
` Ok(key_expr) => key_expr,`

Reply:
I don't see how, we need to break the execution of the function with the return after throwing an exception on the jvm.